vm/actor/src/builtin/market/state.rs

```diff
@@ -54,21 +55,39 @@ pub struct State {
 }
 
 impl State {
-    pub fn new(empty_arr: Cid, empty_map: Cid, empty_mset: Cid) -> Self {
-        Self {
-            proposals: empty_arr,
-            states: empty_arr,
-            pending_proposals: empty_map,
-            escrow_table: empty_map,
-            locked_table: empty_map,
+    pub fn new<BS: BlockStore>(store: &BS) -> Result<Self, Box<dyn StdError>> {
+        let empty_proposals_array =
+            Amt::<(), BS>::new_with_bit_width(store, PROPOSALS_AMT_BITWIDTH)
+                .flush()
+                .map_err(|e| format!("Failed to create empty proposals array: {}", e))?;
+        let empty_states_array = Amt::<(), BS>::new_with_bit_width(store, STATES_AMT_BITWIDTH)
+            .flush()
+            .map_err(|e| format!("Failed to create empty states array: {}", e))?;
+
+        let empty_pending_proposals_map = make_map::<_, ()>(store)
+            .flush()
+            .map_err(|e| format!("Failed to create empty pending proposals map state: {}", e))?;
+        let empty_balance_table = BalanceTable::new(store)
+            .root()
+            .map_err(|e| format!("Failed to create empty balance table map: {}", e))?;
+
+        let empty_deal_ops_hamt = SetMultimap::new(store)
+            .root()
+            .map_err(|e| format!("Failed to create empty multiset: {}", e))?;
+        Ok(Self {
+            proposals: empty_proposals_array,
+            states: empty_states_array,
+            pending_proposals: empty_pending_proposals_map,
+            escrow_table: empty_balance_table,
+            locked_table: empty_balance_table,
             next_id: 0,
-            deal_ops_by_epoch: empty_mset,
+            deal_ops_by_epoch: empty_deal_ops_hamt,
             last_cron: EPOCH_UNDEFINED,
 
             total_client_locked_colateral: TokenAmount::default(),
             total_provider_locked_colateral: TokenAmount::default(),
             total_client_storage_fee: TokenAmount::default(),
-        }
+        })
     }
 
```

Comment:
I can't comment anywhere that isn't changed, but the assertions in this file should probably be changed to state errors, as they did

Reply:
Yeah, good call.